lib/Doctrine/DBAL/Migrations/Provider/LazySchemaDiffProvider.php

```diff
@@ -32,8 +33,23 @@ class LazySchemaDiffProvider implements SchemaDiffProviderInterface
     /** @var SchemaDiffProviderInterface */
     private $originalSchemaManipulator;
 
-    public function __construct(LazyLoadingValueHolderFactory $proxyFactory, SchemaDiffProviderInterface $originalSchemaManipulator)
+    /** @var  Configuration */
+    private $proxyConfig;
+
+    public function __construct(LazyLoadingValueHolderFactory $proxyFactory=null, SchemaDiffProviderInterface $originalSchemaManipulator)
     {
+        if ($proxyFactory === null ) {
+            $this->proxyConfig = new Configuration();
+            $proxyFolder = __DIR__ . DIRECTORY_SEPARATOR . 'SchemaProxy';
```

Comment:
I suggest using a name that is more clear than `SchemaProxy` (`DoctrineMigrationShemaProxy`, for example)